js/collaboration/collaboration.js

```diff
@@ -1,18 +1,55 @@
+// Copyright (c) 2024 Ajeet Pratap Singh
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the The GNU Affero General Public
+// License as published by the Free Software Foundation; either
+// version 3 of the License, or (at your option) any later version.
+//
+// You should have received a copy of the GNU Affero General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, 51 Franklin Street, Suite 500 Boston, MA 02110-1335 USA
+
+
+// Collaboration
+
 /* eslint-disable no-undef */
-import Y, { Doc } from "yjs";
-import SocketIOProvider from "y-socket.io";
 
-// const Y = require("yjs");
-// const SocketIOProvider = require("y-socket.io");
+const RETRIES = 5;
+const DELAY_DURATION = 2000;
+let attempts = 0;
+
+function stopConnection(socket) {
+    if(attempts >= RETRIES){
+        console.log("Maximum calls to make connection exceeded. Stopped making calls");
+        socket.disconnect();
+    }
+}
+
+function makeConnection(){
+
+    // connect to the local server
+    const socket = io("http://localhost:8080/");
```

Comment:
I presume this URL is for testing purposes only?

Reply:
Exactly! We will replace it later for the production.